gsigner/src/cli/commands.rs

```diff
@@ -0,0 +1,183 @@
+    Generate {
+        #[arg(short, long, help = "Storage directory (default: memory only)")]
+        storage: Option<PathBuf>,
+    },
+    #[command(about = "Sign data with a secp256k1 private key")]
+    Sign {
+        #[arg(short, long, help = "Public key (hex)")]
+        public_key: String,
+        #[arg(short, long, help = "Data to sign (hex)")]
+        data: String,
+        #[arg(short, long, help = "Storage directory")]
+        storage: Option<PathBuf>,
+        #[arg(short = 'c', long, help = "Contract address for EIP-191 signing (hex)")]
+        contract: Option<String>,
+    },
+    #[command(about = "Verify a secp256k1 signature")]
+    Verify {
+        #[arg(short, long, help = "Public key (hex)")]
+        public_key: String,
+        #[arg(short, long, help = "Data that was signed (hex)")]
+        data: String,
+        #[arg(short, long, help = "Signature (hex)")]
+        signature: String,
+    },
+    #[command(about = "Get Ethereum address from public key")]
+    Address {
+        #[arg(short, long, help = "Public key (hex)")]
+        public_key: String,
+    },
+    #[command(about = "List all keys in storage")]
+    List {
+        #[arg(short, long, help = "Storage directory")]
+        storage: Option<PathBuf>,
+    },
+}
+
+/// Sr25519 subcommands
+#[derive(Subcommand, Debug, Clone)]
+pub enum Sr25519Commands {
+    #[command(about = "Generate a new sr25519 keypair")]
+    Generate {
+        #[arg(short, long, help = "Storage directory (default: memory only)")]
+        storage: Option<PathBuf>,
+    },
+    #[command(about = "Import sr25519 key from SURI (//Alice, mnemonic, etc.)")]
+    Import {
+        #[arg(
+            short = 'u',
+            long,
+            help = "SURI string (e.g., //Alice, //Alice//stash, mnemonic phrase)"
+        )]
+        suri: String,
+        #[arg(short = 'w', long, help = "Password for SURI derivation")]
+        password: Option<String>,
+        #[arg(short, long, help = "Storage directory (default: memory only)")]
+        storage: Option<PathBuf>,
+    },
+    #[command(about = "Sign data with a sr25519 private key")]
+    Sign {
+        #[arg(short, long, help = "Public key (hex)")]
+        public_key: String,
+        #[arg(short, long, help = "Data to sign (hex)")]
+        data: String,
+        #[arg(short, long, help = "Storage directory")]
+        storage: Option<PathBuf>,
+        #[arg(short = 'c', long, help = "Signing context")]
+        context: Option<String>,
+    },
+    #[command(about = "Verify a sr25519 signature")]
+    Verify {
+        #[arg(short, long, help = "Public key (hex)")]
+        public_key: String,
+        #[arg(short, long, help = "Data that was signed (hex)")]
+        data: String,
+        #[arg(short, long, help = "Signature (hex)")]
+        signature: String,
+        #[arg(short = 'c', long, help = "Signing context")]
+        context: Option<String>,
+    },
+    #[command(about = "Get SS58 address from public key")]
```

Comment:
we may want to also accept here network prefix for proper ss58 printing (vara = 137, substrate basic = 42 etc)